patches/net-url_request-url_request_http_job.cc.patch

```diff
@@ -10,7 +10,15 @@ index f5e754f4ea0288a685a6932b00f692e3c6638621..d7da91c3607d205fd19cae3be369d4e5
      cookie_store->GetCookieListWithOptionsAsync(
          request_->url(), options,
          base::BindOnce(&URLRequestHttpJob::SetCookieHeaderAndStart,
-@@ -770,6 +771,7 @@ void URLRequestHttpJob::SaveCookiesAndNotifyHeadersComplete(int result) {
+@@ -600,6 +601,7 @@ void URLRequestHttpJob::SetCookieHeaderAndStart(
+ 
+   bool can_get_cookies =
+       (request_info_.privacy_mode == PRIVACY_MODE_DISABLED && CanGetCookies());
++  BRAVE_SETCOOKIEHEADERANDSTART
```

Comment:
this patch can probably be eliminated by changing the subclass of `URLRequestHttpJob` and overriding `CanGetCookies`

Reply:
`CanGetCookies` probably cannot be overridden since it isn't virtual, but I guess it's possible to create a new version of this function or a new method, perhaps with the preprocessor.

Reply:
It seems that putting this in `url_request_http_job.cc` is sufficient to remove the patch.
```c
+#define CanGetCookies() \
+  (CanGetCookies() || CanUseEphemeralStorage(this))
+
```

Then, of course, `#undef CanGetCookies` at the end of the file.

Reply:
@mrobinson that would also work, but I think you still have the problem discussed in slack of differentiating between block 3p and block all

Reply:
@mrobinson that is a good idea though and I think we can still use it per the dm discussion with `|| true`

Reply:
Done